### DISCUSSION TITLE
Error creating a new image

### DISCUSSION THREAD

**Discussion by decenttp** [ltsp/ltsp#865] [Q&A]
I am trying to create a new client image using symlinked virtual image of a Zorin 16 Pro on trying to do the same getting following error: 

ltsp image zorin

LTSP command failed: blkid -po export /srv/ltsp/zorin.img

Aborting LTSP

I don't know how to mount /srv/ltsp/zorin.img

Have created Zorin 15 image in the past without issues. Any ideas what could be wrong?

Thanks

**Comment by alkisg** [comment]
Make sure the image is a -flat.vmdk, not a .vdi.
For example:

```
# blkid -po export /srv/vm/jammy-mate/jammy-mate-flat.vmdk 
DEVNAME=/srv/vm/jammy-mate/jammy-mate-flat.vmdk
PTUUID=7e6cffe9-0434-4ca9-8095-8653afb40c8e
PTTYPE=gpt
```

**Reply by decenttp** [reply]
Thanks for your reply and it was indeed due to the reason you had mentioned. I tried setting up a new image with flat vmdk but getting a different error now: 

ln -s "/home/admin/Downloads/zpro/zpro/zpro-flat.vmdk" /srv/ltsp/zorinpro.img
ltsp image zorinpro
Running: losetup -rP /dev/loop4 /srv/ltsp/zorinpro.img
Running: mount -t tmpfs -o mode=0755 tmpfs /tmp/tmp.9JcFJRKovO/tmpfs
Running: mount -t vfat -o ro /dev/loop4p1 /tmp/tmp.9JcFJRKovO/tmpfs/0/looproot
Running: mount -t overlay -o upperdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/up,lowerdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/looproot,workdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/work /tmp/tmp.9JcFJRKovO/tmpfs /tmp/tmp.9JcFJRKovO/root/
mount: /tmp/tmp.9JcFJRKovO/root: wrong fs type, bad option, bad superblock on /tmp/tmp.9JcFJRKovO/tmpfs, missing codepage or helper program, or other error.
LTSP command failed: mount -t overlay -o upperdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/up,lowerdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/looproot,workdir=/tmp/tmp.9JcFJRKovO/tmpfs/0/work /tmp/tmp.9JcFJRKovO/tmpfs /tmp/tmp.9JcFJRKovO/root/
Aborting ltsp

Image is  working fine in virtual box 7.0

**Reply by decenttp** [reply]
Size of the flat vmdk is 40gb

**Comment by alkisg** [comment]
And what is the output of this now?

    sudo blkid -po export /srv/ltsp/zorinpro.img

**Reply by decenttp** [reply]
blkid -po export /srv/ltsp/zorinpro.img
DEVNAME=/srv/ltsp/zorinpro.img
PTUUID=a32e6e71
PTTYPE=dos

**Comment by alkisg** [comment]
That doesn't look like an ext4 partition.
Boot the VM, run `sudo fdisk -l` inside it, and paste the output.
LTSP supports flat.vmdk disks with simple ext4 partitions. No encryption, no lvm.

**Reply by decenttp** [reply]
sudo fdisk -l

Disk /dev/sda: 40 GiB, 42949672960 bytes, 83886080 sectors
Disk model: VBOX HARDDISK
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xa32e6e71

Device     Boot   Start      End  Sectors  Size Id Type
/dev/sda1  *       2048  1050623  1048576  512M  b W95 FAT32
/dev/sda2       1052670 83884031 82831362 39.5G  5 Extended
/dev/sda5       1052672 83884031 82831360 39.5G 83 Linux

**Comment by alkisg** [comment]
Yeah create a single ext4 partition instead of a separate /boot partition and an extended+logical partition.

**Reply by decenttp** [reply]
Thanks for pointing that out. I am resetting the like you told.

**Comment by decenttp** [comment]
Using single ext4 partition resolved the issue for me and everything worked fine. For some reason this image is not mounting the /home folder at startup and so the users don't see their folders on logging in. What should i be checking?

**Comment by alkisg** [comment]
Install ltsp inside the VM.

**Reply by decenttp** [reply]
I only installed ltsp package but that didn't resolve the issue. Should i install all of the ltsp server components? : 

apt install --install-recommends ltsp ltsp-binaries dnsmasq nfs-kernel-server openssh-server squashfs-tools ethtool net-tools epoptes
gpasswd -a administrator epoptes

**Comment by alkisg** [comment]
After logging in on a client, what's the output of the following commands?

```shell
dpkg -l ltsp sshfs
mount | grep home
```

**Reply by decenttp** [reply]
I rebooted the ltsp server and after that users can see the folders but most of the applications won't launch. They show permission error writing to /home/USER folder 

While booting up i still see the mount failed error for /home

Output of the two commands : 

![image](https://github.com/ltsp/ltsp/assets/88873876/43c56dd7-0c2e-499f-9f41-2910b694b8dd)

mount | grep home
nlinux@server: on /home/nlinux type fuse.sshfs
(rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other)
nlinuxadmin@server: on /home/nlinuxadmin type fuse.sshfs
(rw,nosuid,nodev,relatime,user_id=0,group_id=0,default_permissions,allow_other)
gvfsd-fuse on /home/nlinuxadmin/.cache/gvfs type fuse.gvfsd-fuse
(rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
portal on /home/nlinuxadmin/.cache/doc type fuse.portal
(rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)

**Comment by alkisg** [comment]
I think it'll be faster if you come in the https://ltsp.org/guides/chat-room/ for live troubleshooting.

**Reply by decenttp** [reply]
As discussed I created a new image using Ubuntu Mate but it comes with snap pre-installed and I see the exact same issue with snap pre-installed tools like firefox.

**Comment by alkisg** [comment]
See https://ltsp.org/guides/snap/

**Reply by decenttp** [reply]
Hi, skype is only available through snap now. Any workarounds to get it to work apart from running it in the browser?
